src/server/client.rs

```diff
@@ -136,6 +140,52 @@ pub async fn client_rpc_loop<
     }
 }
 
+async fn handle_wait_for_jobs_message(state_ref: &StateRef, selector: Selector) -> ToClientMessage {
+    let (receivers, mut response) = {
+        let mut state = state_ref.get_mut();
+        let job_ids: Vec<JobId> = get_job_ids(&state, selector);
+
+        let mut response = WaitForJobsResponse::default();
+        let mut receivers = vec![];
+
+        for job_id in job_ids {
+            match state.get_job_mut(job_id) {
+                Some(job) => {
+                    if job.is_terminated() {
+                        if job.counters.has_unsuccessful_tasks() {
+                            response.failed += 1;
```

Comment:
This does not seem right, has_unsucessful_tasks counts also "canceled"; however, "response.failed" looks like it counts only fail states (so some error messages later may be wrong?)

Reply:
`failed` counts both failed and canceled tasks. It's purpose is to cause `wait` to exit with a non-success error code if `failed > 0`.

Reply:
Then it is incorrectly named, "failed" has a quite precise meaning in HQ.

Reply:
So should `wait` also track canceled jobs? Or just rename it to e.g. `unsuccessful`?